File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/AbpEventBusBoxesOptions.cs
using System;

namespace Volo.Abp.EventBus.Boxes;

public class AbpEventBusBoxesOptions
{
    /**
     * Default: 6 hours
    */
    public TimeSpan CleanOldEventTimeIntervalSpan;// { get; set; }

    /**
     * Default: 1000
    */
    public int InboxWaitingEventMaxCount;// { get; set; }

    /**
     * Default: 1000
    */
    public int OutboxWaitingEventMaxCount;// { get; set; }

    /**
     * Period time of <see cref="InboxProcessor"/> and <see cref="OutboxSender"/>
     * Default: 2 seconds
    */
    public TimeSpan PeriodTimeSpan;// { get; set; }

    /**
     * Default: 15 seconds
    */
    public TimeSpan DistributedLockWaitDuration;// { get; set; }

    /**
     * Default: 2 hours
    */
    public TimeSpan WaitTimeToDeleteProcessedInboxEvents;// { get; set; }

    /**
     * Default: true
    */
    public boolean BatchPublishOutboxEvents;// { get; set; }

    public AbpEventBusBoxesOptions()
    {
        CleanOldEventTimeIntervalSpan = TimeSpan.FromHours(6);
        InboxWaitingEventMaxCount = 1000;
        OutboxWaitingEventMaxCount = 1000;
        PeriodTimeSpan = TimeSpan.FromSeconds(2);
        DistributedLockWaitDuration = TimeSpan.FromSeconds(15);
        WaitTimeToDeleteProcessedInboxEvents = TimeSpan.FromHours(2);
        BatchPublishOutboxEvents = true;
    }
}

File: abp-frameworks/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/AbpEventBusBoxesOptions.java
package com.ueueo.eventbus.distributed;

import lombok.Data;

import java.time.Duration;

@Data
public class AbpEventBusBoxesOptions {
    /**
     * Default: 6 hours
     */
    private Duration cleanOldEventTimeIntervalSpan = Duration.ofHours(6);

    /**
     * Default: 1000
     */
    private int inboxWaitingEventMaxCount = 1000;

    /**
     * Default: 1000
     */
    private int outboxWaitingEventMaxCount = 1000;

    /**
     * Period time of <see cref="InboxProcessor"/> and <see cref="OutboxSender"/>
     * Default: 2 seconds
     */
    private Duration periodTimeSpan = Duration.ofSeconds(2);

    /**
     * Default: 15 seconds
     */
    private Duration distributedLockWaitDuration = Duration.ofSeconds(15);

    /**
     * Default: 2 hours
     */
    private Duration waitTimeToDeleteProcessedInboxEvents = Duration.ofHours(2);

    /**
     * Default: true
     */
    private boolean batchPublishOutboxEvents = true;

    public AbpEventBusBoxesOptions() {
    }
}
